# DHI-only patch (not upstream). Targets the Debian gtk+3.0 source package:
#   https://sources.debian.org/src/gtk+3.0/3.24.49-3/gtk/gen-gtk-gresources-xml.py/
#
# gtk.gresources.xml uses preprocess="to-pixdata" for theme PNGs. glib-compile-resources
# then invokes gdk-pixbuf-pixdata under debian/rules' temporary HOME/XDG overrides,
# which fails PNG decode in the hermetic builder even though direct PNG embed works.
--- a/gtk/gen-gtk-gresources-xml.py
+++ b/gtk/gen-gtk-gresources-xml.py
@@ -23,7 +23,7 @@ xml += '''
 '''
 
 for f in get_files('theme/Adwaita/assets', '.png'):
-  xml += '    <file preprocess=\'to-pixdata\'>theme/Adwaita/assets/{0}</file>\n'.format(f)
+  xml += '    <file>theme/Adwaita/assets/{0}</file>\n'.format(f)
 
 xml += '\n'
 
@@ -38,7 +38,7 @@ xml += '''
 '''
 
 for f in get_files('theme/HighContrast/assets', '.png'):
-  xml += '    <file preprocess=\'to-pixdata\'>theme/HighContrast/assets/{0}</file>\n'.format(f)
+  xml += '    <file>theme/HighContrast/assets/{0}</file>\n'.format(f)
 
 xml += '\n'
 
